nixos/modules/config/nsswitch.nix

```diff
@@ -93,14 +73,11 @@ with lib;
   ];
 
   config = {
-    assertions = [
-      {
-        # Prevent users from disabling nscd, with nssModules being set.
-        # If disabling nscd is really necessary, it's still possible to opt out
-        # by forcing config.system.nssModules to [].
-        assertion = config.system.nssModules.path != "" -> config.services.nscd.enable;
-        message = "Loading NSS modules from system.nssModules (${config.system.nssModules.path}), requires services.nscd.enable being set to true.";
-      }
+    # Provide NSS modules at a platform-specific path in /run
+    systemd.tmpfiles.rules = let
+      glibcPlatform = "${if pkgs.stdenv.hostPlatform.is64bit then "64" else "32"}-${pkgs.glibc.version}";
+    in [
+      "L+ /run/nss-modules-${glibcPlatform} - - - - ${config.system.nssModules.path}"
```

Comment:
So this will "leak" symlinks over time as glibc versions change? Couldn't this be part of /run/current-system?

Reply:
This will create symlinks to nix store paths there (until it's rebooted). Nix might eventually garbage collect them away, I'm not sure if `/run` is checked. In that case, the NSS lookup will fail using that NSS module, which is probably still better than segfaulting ;-)

Having these NSS moduels inside /run/current-system would mean one would need to keep multiple glibc versions around as part of the system closure.

As this is only meant to prevent breakage of still running old versions of binaries, compiled against an old glibc (and even those will use nscd by default), I'd consider this to be not much of a problem.